cpp/test/neighbors/ann_ivf_flat.cuh

```diff
@@ -546,12 +562,14 @@ const std::vector<AnnIvfFlatInputs<int64_t>> inputs = {
   {20, 100000, 16, 10, 20, 1024, cuvs::distance::DistanceType::L2Expanded, true},
   {1000, 100000, 16, 10, 20, 1024, cuvs::distance::DistanceType::L2Expanded, true},
   {10000, 131072, 8, 10, 20, 1024, cuvs::distance::DistanceType::L2Expanded, false},
+  {3, 20, 1024, 3, 3, 4, cuvs::distance::DistanceType::CosineExpanded, false},
```

Comment:
Why are we testing such a limited set of cases here? Ideally we would remove the metric from these params and always run the given params for all supported metrics.